compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -896,7 +896,11 @@ impl<'a> Parser<'a> {
         let has_lifetime = self.token.is_lifetime() && self.look_ahead(1, |t| t != &token::Colon);
         let lifetime = has_lifetime.then(|| self.expect_lifetime()); // For recovery, see below.
         let (borrow_kind, mutbl) = self.parse_borrow_modifiers(lo);
-        let expr = self.parse_prefix_expr(None);
+        let expr = if self.token.is_range_seperator() {
+            self.parse_prefix_range_expr(None)
+        } else {
+            self.parse_prefix_expr(None)
+        };
```

Comment:
My immediate expectation would be that `parse_prefix_expr` includes `parse_prefix_range_expr`. Are there any cases where we use `parse_prefix_expr` and we want prefix range expressions to be an error? If yes, make sure there's a test for that; if no, then it's probably better to move the range separator check/parse inside `parse_prefix_expr`.

I've not looked at this code much, but the reason it might matter is for other operator precedence reasons; e.g. what's the correct result of parsing `+..1` or `1+..1`.

Reply:
I think this separation of prefix range parsing from other prefix expr parsing is done to prevent expressions such as `-..1` or `!..1` from being parsed. Other unary operators can be stacked (may not be typeable), but you cannot have `..` or `..=` after a unary operator.